api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -0,0 +1,156 @@
+	
+	protected OrderService orderService;
+	
+	protected OrderSetService orderSetService;
+	
+	protected ConceptService conceptService;
+	
+	protected static final String ORDER_SET = "org/openmrs/api/include/OrderSetServiceTest-general.xml";
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+	
+	/**
+	 * Run this before each unit test in this class. The "@Before" method in
+	 * {@link BaseContextSensitiveTest} is run right before this method.
+	 *
+	 * @throws Exception
+	 */
+	@Before
+	public void runBeforeAllTests() throws Exception {
+		if (null == orderSetService) {
+			orderSetService = Context.getOrderSetService();
+		}
+		if (null == orderService) {
+			orderService = Context.getOrderService();
+		}
+		if (null == conceptService) {
+			conceptService = Context.getConceptService();
+		}
+	}
+	
+	@After
+	public void revertToDefaultLocale() throws Exception {
+		Context.setLocale(Locale.US);
+	}
+	
+	/**
+	 * Updates the search index to clean up after each test.
+	 *
+	 * @see org.openmrs.test.BaseContextSensitiveTest#updateSearchIndex()
+	 */
+	@Before
+	@Override
+	public void updateSearchIndex() {
+		super.updateSearchIndex();
+	}
+	
+	/**
+	 * Updates the search index after executing each dataset.
+	 *
+	 * @see org.openmrs.test.BaseContextSensitiveTest#executeDataSet(org.dbunit.dataset.IDataSet)
+	 */
+	@Override
+	public void executeDataSet(IDataSet dataset) throws Exception {
+		super.executeDataSet(dataset);
+		updateSearchIndex();
+	}
+	
+	@Test
+	@Verifies(value = "should save OrderSet", method = "save(OrderSet)")
+	public void shouldSaveOrderSet() throws Exception {
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetBuilder(false, false);
+		OrderSet orderSetObj = orderSetService.save(orderSet);
+		assertNotNull("OrderSet has a order_set_id", orderSetObj.getId().intValue());
+	}
+	
+	@Test
+	@Verifies(value = "should update existing OrderSet", method = "save(OrderSet)")
+	public void shouldSaveAndUpdateOrderSet() throws Exception {
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetBuilder(false, false);
+		orderSet.setRetired(true);
+		OrderSet orderSetObj = orderSetService.save(orderSet);
+		assertEquals("OrderSet should contain orderSetmembers", 1, orderSetObj.getOrderSetMembers().size());
+		
+		orderSetObj.setOperator(OrderSet.Operator.ONE);
+		orderSetObj.setDescription("Test Order Set Description Updated");
+		orderSetObj.setOrderSetMembers(new HashSet<OrderSetMember>());
+		orderSetObj = orderSetService.save(orderSetObj);
+		assertNotNull("OrderSet has a order_set_id", orderSetObj.getId().intValue());
+		assertEquals("OrderSet has updated description", "Test Order Set Description Updated", orderSetObj.getDescription());
+		assertEquals("OrderSet has updated operator", "ONE", orderSetObj.getOperator().toString());
+		assertEquals("OrderSet should not contain orderSetmembers", 0, orderSetObj.getOrderSetMembers().size());
```

Comment:
Can you also assert that dateChanged and changedBy have been set? You will also need sanity checks before saving to ensure they are null